github/compliance/model.py

```diff
@@ -174,8 +191,11 @@ def result_groups(self) -> tuple[ScanResultGroup]:
             return ()
 
         for result in self.results:
-            artifact = cnudie.iter.artifact_from_node(result.scanned_element)
-            group_name = f'{result.scanned_element.component.name}:{artifact.name}'
+            if isinstance(result.scanned_element, (cnudie.iter.ResourceNode, cnudie.iter.SourceNode)): # noqa: E501
```

Comment:
since this is now checked for the third time, why not make this a property? e.g. `scanned_element_is_ocm_artefact`

Reply:
What about the `_target_*` functions, such as `_target_repository` or `_target_assignees`?
Should we still pass target and manually check for `isinstance(target, (SN, RN))` or alter to:
```python
def _scan_result_repository(result: ScanResult):
  if result.scanned_element_is_ocm_resource:
    ...
```
?

Reply:
exactly, this deduplication was what I had in mind.